File: src/python/insecure_tempfile.py
"""CWE-377: Insecure temporary file (mktemp / predictable name).

Rule: py/use-of-mktemp, py/insecure-temporary-file
Autofix: typically NOT provided — fix requires switching to mkstemp /
NamedTemporaryFile and propagating the new return signature.
"""

import os
import tempfile


def stage(data: bytes) -> str:
    name = tempfile.mktemp(suffix=".bin")
    with open(name, "wb") as f:
        f.write(data)
    return name


def workdir() -> str:
    path = "/tmp/work-" + str(os.getpid())
    os.mkdir(path)
    return path

File: src/javascript/insecure-tempfile.js
// CWE-377: Insecure temporary file (predictable name / TOCTOU).
// Rule: js/insecure-temporary-file
// Autofix: typically NOT provided — fix requires switching to a safe API
// (e.g. mkdtempSync / tmp.file with discardDescriptor: false) and threading
// the cleanup logic through the call-sites.

const fs = require('fs');
const os = require('os');
const path = require('path');

function stageUpload(buf) {
  const name = path.join(os.tmpdir(), 'upload-' + Date.now() + '.bin');
  fs.writeFileSync(name, buf);
  return name;
}

function newWorkdir() {
  const dir = path.join('/tmp', 'work-' + process.pid);
  fs.mkdirSync(dir);
  return dir;
}

module.exports = { stageUpload, newWorkdir };
